src/broker/runat.c

```diff
@@ -244,7 +244,7 @@ static void stdio_cb (flux_subprocess_t *p, const char *stream)
     const char *line;
```

Comment:
commit message and PR title: I'd probably call this "improve read API" since it's really about the API change not refactoring.

Reply:
good point, re-pushed with that tweaked commit message